fsspec/implementations/zip.py

```diff
@@ -133,3 +141,16 @@ def _open(
 
     def ukey(self, path):
         return tokenize(path, self.fo, self.protocol)
+
+    def _all_dirnames(self, paths):
```

Comment:
Might want to extract this to a separate function, so as not to pollute the FS namespace. Since it's marked private, maybe that's unimportant.

Reply:
I originally had this function as part of test_zip.py, because I needed a way to generate all the intermediate directories. Then I realized that we needed the same behavior for the class as well. 

I am happy to move it where you prefer. Here are the options I can think:
   1. Private at ZipFileSystem (as it is)
   1. Keep it private but move it up to AbstractFileSystem, and make the "root_maker" a function parameter.
   1. Make it public and place it on utils.py

Reply:
In the end, I say leave it here. We can always move it later if need be.